manifests/instance/tls.pp

```diff
@@ -0,0 +1,180 @@
+# @summary Configure TLS for an instance
+#
+# Requires LDAPI to be enabled
+#
+# @author https://github.com/simp/pupmod-simp-ds389/graphs/contributors
+#
+define ds389::instance::tls (
+  String[2]                                 $root_dn,
+  Stdlib::Absolutepath                      $root_pw_file,
+  Variant[Boolean, Enum['disabled','simp']] $ensure        = simplib::lookup('simp_options::pki', { 'default_value' => false}),
+  Simplib::Port                             $port          = 636,
+  String[1]                                 $source        = simplib::lookup('simp_options::pki', { 'default_value' => "/etc/pki/simp/${module_name}_${title}"}),
+  Stdlib::Absolutepath                      $cert          = "/etc/pki/simp_apps/${module_name}_${title}/x509/public/${facts['fqdn']}.pub",
+  Stdlib::Absolutepath                      $key           = "/etc/pki/simp_apps/${module_name}_${title}/x509/private/${facts['fqdn']}.pem",
+  Stdlib::Absolutepath                      $cafile        = "/etc/pki/simp_apps/${module_name}_${title}/x509/cacerts/cacerts.pem",
+  Ds389::ConfigItems                        $dse_config    = simplib::dlookup('ds389::instance::pki', 'dse_config', { 'default_value' => {} }),
+  String[16]                                $token         = simplib::passgen("ds389_${title}_pki", { 'length' => 32 }),
+  String[1]                                 $service_group = 'dirsrv'
+) {
+  assert_private()
+
+  $_instance_name = split($title, /^(dirsrv@)?slapd-/)[-1]
+
+  $_default_dse_config = {
+    'cn=encryption,cn=config' => {
+      'allowWeakCipher'               => 'off',
+      'allowWeakDHParam'              => 'off',
+      'nsSSL2'                        => 'off',
+      'nsSSL3'                        => 'off',
+      'nsSSLClientAuth'               => 'required',
+      'nsTLS1'                        => 'on',
+      'nsTLSAllowClientRenegotiation' => 'on',
+      'sslVersionMax'                 => 'TLS1.2',
+      'sslVersionMin'                 => 'TLS1.2'
+    },
+    'cn=config'               => {
+      'nsslapd-ssl-check-hostname' => 'on',
+      'nsslapd-validate-cert'      => 'on',
+      'nsslapd-minssf'             => 256
+    }
+  }
+
+  $_required_dse_config = {
+    'cn=config' => {
+      'nsslapd-security'   => 'on',
+      'nsslapd-securePort' => $port
+    }
+  }
+
+  if $ensure == 'disabled' {
```

Comment:
Do you need to `ensure  =>absent` the `selinux_port` below if disabled?  Thinking of the case in which this resource is created by a `ds389::instance` and then that instance is disabled with `ensure =>absent`, not if `simp_options::pki` gets set to false.  In other words, worried about resource hanging around when a `ds389::instance` is removed.

Reply:
Ah yeah, good call.